code/ConvertToStandardPath_SubmarineCable.py

```diff
@@ -295,38 +285,95 @@ def calculate_distance_between_cable_landing_points(wktToCities, cableIdToWkt):
         
+        
         # calculate each city pair distance on upper graph
         for city_pair in city_pairs:
             # format the start city and end city and verify them on the graph.
-            start_city = (floatFormatter(
-                city_pair[0][2]), floatFormatter(city_pair[0][1]))
-            end_city = (floatFormatter(
-                city_pair[1][2]), floatFormatter(city_pair[1][1]))
+            start_city, end_city = city_pair
+            start_city_name, start_city_state, start_city_country, start_city_lati, start_city_longti = start_city
+            end_city_name, end_city_state, end_city_country, end_city_lati, end_city_longti = end_city
+
+            start_city_coord = (start_city_lati, start_city_longti)
+            end_city_coord = (end_city_lati, end_city_longti)
             graph = verify_graph_with_cities(
-                graph, start_city, end_city, cable_id)
+                graph, start_city_coord, end_city_coord, cable_id)
             
-            if (start_city in graph and end_city in graph):
+            if (start_city_coord in graph and end_city_coord in graph):
                 try:
                     shortest_path = nx.shortest_path(
-                        graph, source=start_city, target=end_city, weight='weight')
+                        graph, source=start_city_coord, target=end_city_coord, weight='weight')
                     shortest_path_length = nx.shortest_path_length(
-                        graph, source=start_city, target=end_city, weight='weight')
-                    print(
-                        f"city {city_pair[0][0]}, {start_city}, city {city_pair[1][0]}, {end_city} on {cable_id} has shortest_path_length {shortest_path_length} consist of {shortest_path}")
+                        graph, source=start_city_coord, target=end_city_coord, weight='weight')
+                    # print(
+                    #      f"city {start_city_name}, {start_city_state}, {start_city_country}, {start_city_coord}, city {end_city_name}, {end_city_state}, {end_city_country}, {end_city_coord} on {cable_id} has shortest_path_length {shortest_path_length} consist of {shortest_path}")
+                    submarine_standard_paths.append(((start_city_name, start_city_state, start_city_country, 
+                               end_city_name, end_city_state, end_city_country, 
+                               shortest_path_length, list_to_linestring(shortest_path))))
                 except NetworkXNoPath:
                     continue
                     # print(
-                    #     f"city {city_pair[0][0]}, {start_city}, city {city_pair[1][0]}, {end_city} on {cable_id} does not have shortest_path_length")
+                    #     f"city {start_city_name}, {start_city_state}, {start_city_country}, {start_city_coord}, city {end_city_name}, {end_city_state}, {end_city_country}, {end_city_coord} on {cable_id} does not have shortest_path_length")
             else:
```

Comment:
Why could this happen? Is this something we need to investigate or is okay due to the nature of the data (missing data in one place vs another).

It does look like the cities and the graphs are constructed from this one table, so maybe this shouldn't happen at all?

Reply:
Yes, this part of code will not be triggered. Just use it to verify the verify_graph_with_cities works as expection.